src/Std/Tactic/BVDecide/Normalize/BitVec.lean

```diff
@@ -206,5 +206,17 @@ theorem BitVec.max_ult' (a : BitVec w) : (BitVec.ult (-1#w) a) = false := by
 
 attribute [bv_normalize] BitVec.replicate_zero_eq
 
+@[bv_normalize]
+theorem BitVec.ofBool_getLsbD (a : BitVec w) (i : Nat) :
+    BitVec.ofBool (a.getLsbD i) = a.extractLsb' i 1 := by
+  ext j
+  simp
+
+@[bv_normalize]
+theorem BitVec.ofBool_getElem (a : BitVec w) (i : Nat) (h : i < w) :
+    BitVec.ofBool a[i] = a.extractLsb' i 1 := by
+  rw [show a[i] = a.getLsbD i from rfl]
```

Comment:
There should probably exist a lemma for this? If it doesn't I think we should add it?

Reply:
You're right, I didn't find it cause my master was outdated.

I did golf the proof of the lemma while I was at it: it used to involve `simp`, but `rfl` is enough.